### PR TITLE
Added section for promoted packages in the back-office.

### PR DESCRIPTION
**Pull request by AndyButland**
This PR adds an additional panel to the Umbraco back-office Packages section, so between "Popular" and "New releases", we can have a "Promoted" section.  The intention here is to have the option to highlight HQ, technical partner and selected community packages.

To Test:

- Check with the PR in place that the back-office presents a new panel showing the HQ packages.
- Via the our back-office, remove the "Is Promoted?" flag from one, and check the display is updated (given a little time for index updates and cache invalidation).
- Then remove both and ensure the panel and the header is removed.

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js [hunk 1]
```diff
@@ -36,7 +36,21 @@ function ourPackageRepositoryResource($q, $http, umbDataFormatter, umbRequestHel
...
-       
+
+        getPromoted: function (maxResults, category) {
+
+            if (maxResults === undefined) {
+                maxResults = 20;
+            }
+            if (category === undefined) {
+                category = "";
+            }
```
Comment (bjarnef) [inline src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js:47]: How about this instead:
https://github.com/umbraco/Umbraco-CMS/blob/8eb3ef65f686619d09af478bd6fd563fc90580b2/src/Umbraco.Web.UI.Client/src/common/resources/log.resource.js#L65-L76

```
var defaults = {
       pageSize: 10,
       pageNumber: 1,
};

if (options === undefined) {
   options = {};
}

//overwrite the defaults if there are any specified
Utilities.extend(defaults, options);
//now copy back to the options we will use
options = defaults;
```

or like this which extend the original `config` object:
https://github.com/umbraco/Umbraco-CMS/blob/5bfab13dc5a268714aad2426a2b68ab5561a6407/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/colorpicker.controller.js#L4-L13

File: src/Umbraco.Web.UI.Client/src/views/packages/views/repo.controller.js [hunk 2]
```diff
@@ -24,6 +24,8 @@
         vm.closeLightbox = closeLightbox;
         vm.search = search;
         vm.installCompleted = false;
+        vm.highlightedPackageCollections = [];
+        vm.labels = [];
```
Comment (bjarnef) [inline src/Umbraco.Web.UI.Client/src/views/packages/views/repo.controller.js:28]: Maybe change this to `vm.labels = {}` instead.

Then we could later do the following:

```
vm.labels.popularPackages  = labels[0];
vm.labels.promotedPackages = labels[1];
```

Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/views/packages/views/repo.controller.js:28]: That's nicer certainly - have updated to use that pattern.

File: src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js [hunk 3]
```diff
@@ -36,7 +36,21 @@ function ourPackageRepositoryResource($q, $http, umbDataFormatter, umbRequestHel
...
+
+            return umbRequestHelper.resourcePromise(
+               $http.get(baseurl + "?pageIndex=0&pageSize=" + maxResults + "&category=" + category + "&order=Popular&version=" + Umbraco.Sys.ServerVariables.application.version + "&onlyPromoted=true"),
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js:50]: I agree that's nicer if this were a brand new feature, but this needs to match with what's accepted at the API endpoint on our.umbraco.com, so I don't think it's worth changing.  Even if we changed it on both ends, we would need to keep the existing as well as introduce the new as of course older back-offices would still use the old parameter name.

File: src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js (hunk 1 above)
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js:47]: Might be nicer I agree, but I'm following existing patterns in this file with the changes, so am more comfortable just continuing with what's working already for the other cases.

**Review by bergmania — APPROVED** [review]

**Pull request merged**